bin/plugin/commands/performance.js

```diff
@@ -187,226 +218,248 @@ function curateResults( testSuite, results ) {
+ * Runs the performance tests on the current ref.
  *
- * @param {string} testSuite                Name of the tests set.
- * @param {string} performanceTestDirectory Path to the performance tests' clone.
- * @param {string} runKey                   Unique identifier for the test run, e.g. `branch-name_post-editor_run-3`.
- *
- * @return {Promise<WPPerformanceResults>} Performance results for the branch.
+ * @param {string} testSuite Name of the tests set.
+ * @param {string} runKey    Unique identifier for the test run, e.g. 'post-editor_ref-name_run-3'.
+ * @return {Promise<WPPerformanceResults>} Performance results for the ref.
  */
-async function runTestSuite( testSuite, performanceTestDirectory, runKey ) {
+async function runTestSuite( testSuite, runKey ) {
 	const resultsFilename = `${ runKey }.performance-results.json`;
 
-	await runShellScript(
-		`npm run test:performance -- ${ testSuite }`,
-		performanceTestDirectory,
-		{
-			...process.env,
-			WP_ARTIFACTS_PATH: ARTIFACTS_PATH,
-			RESULTS_FILENAME: resultsFilename,
-		}
-	);
+	await runShellScript( `npm run test:performance -- ${ testSuite }`, CWD, {
+		...process.env,
+		WP_ARTIFACTS_PATH: ARTIFACTS_PATH,
+		RESULTS_FILENAME: resultsFilename,
+	} );
 
 	return curateResults(
 		testSuite,
-		await readJSONFile( path.join( ARTIFACTS_PATH, resultsFilename ) )
+		readJSONFile( path.join( ARTIFACTS_PATH, resultsFilename ) )
 	);
 }
 
 /**
- * Runs the performances tests on an array of branches and output the result.
+ * Runs the performances tests on an array of git refs and output the result.
+ * Ref can be a branch, commit SHA or a tag.
  *
- * @param {string[]}                    branches Branches to compare
- * @param {WPPerformanceCommandOptions} options  Command options.
+ * @param {string[]}                    refs    Refs to compare.
+ * @param {WPPerformanceCommandOptions} options Command options.
  */
-async function runPerformanceTests( branches, options ) {
+async function runPerformanceTests( refs, options ) {
+	console.time( TIMERS.total );
+
 	const runningInCI = !! process.env.CI || !! options.ci;
-	const TEST_ROUNDS = options.rounds || 1;
+	const testRounds = options.rounds || 1;
 
-	// The default value doesn't work because commander provides an array.
-	if ( branches.length === 0 ) {
-		branches = [ 'trunk' ];
-	}
+	// @ts-ignore
+	let localRef;
 
-	log(
-		formats.title( '\n💃 Performance Tests 🕺\n' ),
-		'\nWelcome! This tool runs the performance tests on multiple branches and displays a comparison table.\n' +
-			'In order to run the tests, the tool is going to load a WordPress environment on ports 8888 and 8889.\n' +
-			'Make sure these ports are not used before continuing.\n'
-	);
+	if ( runningInCI ) {
+		localRef = process.env.GITHUB_HEAD_REF;
+		// @ts-ignore
+	} else if ( await simpleGit( CWD ).checkIsRepo() ) {
+		// @ts-ignore
+		localRef = await simpleGit( CWD ).revparse( {
+			'--abbrev-ref': null,
+			HEAD: null,
+		} );
+	}
```

Comment:
It is not clear to me what we're doing this within the performance command. We're creating a dependency between the performance command and Github CI runner. What if we do this logic outside and pass the "localRef" as a regular branch to compare like we used to do before?

Reply:
Rather than creating a _CI dependency_, I'd argue that it adds _CI support_. When run in CI, the script identifies the local reference and uses it to build the plugin. This is quicker than our previous method, as it circumvents the need to clone the repository and install node modules.

During the refactor, I was fairly confident that this script would not be utilized outside an environment that would be the subject of the test, whether it's Gutenberg or another plugin. TBH, I'm still struggling to see an alternative use-case. If our intention is to build and measure or compare the performance of a specific feature, we can easily switch to the appropriate branch and use the local reference. From your perspective, it seems we want this script to act as a standalone application capable of fetching and performance-testing any references from the origin. While it wouldn't be difficult to include support for that (old) logic in the proposed implementation, I feel I'm yet to understand a compelling use-case example.